### ISSUE TITLE
empty_count: false positive when input argument called `count`

### ISSUE THREAD

**Issue by yas375** [realm/SwiftLint#2012]
### Bug Report

```bash
$ echo "private func upcomingRequests(count: Int) {\nguard count > 0 else {" | swiftlint lint --no-cache --use-stdin --enable-all-rules
Loading configuration from '.swiftlint.yml'
<nopath>:2:7: error: Empty Count Violation: Prefer checking `isEmpty` over comparing `count` to zero. (empty_count)
Done linting! Found 1 violation, 1 serious in 1 file.
```

### Environment

* SwiftLint: 0.24.2
* Installation method used: Homebrew
* `.swiftlint.yml`:

```yml
opt_in_rules:
  - empty_count
```

* No [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)
* Xcode 9.2

This code triggers error message:

```swift
private func upcomingRequests(count: Int) {
    guard count > 0 else {
      return
    }
    // ...
}
```

<img width="932" alt="2018-01-19_2345" src="https://user-images.githubusercontent.com/217368/35181236-e5daf8a6-fd72-11e7-981a-0a3154360cb7.png">

**Comment by marcelofabri** [comment]
Duplicate of #827.

**Comment by yas375** [comment]
@marcelofabri I did search before opening this issue and I did found the one you referenced and that one didn't seem to be the same. That issue is about the fact that `NSMapTable` doesn't have `isEmpty`, so the SwiftLint should warn about switching to `isEmpty` for it. While the issue I'm experiencing is that I'm getting lint error if I have a variable called `count` and if I compare it with zero. Maybe consider updating that PR's description to mention both bugs. Hope it helps. And sorry again if it felt as duplicate.

**Comment by SDGGiesbrecht** [comment]
@yas375, the underlying issue *is* the same. (Though it is admittedly not obvious until further into the comments, so do not worry about it.)

Basically, SwiftLint only has access to the name of the symbol `count`; it has no way of knowing what the symbol describes (i.e. Variable or property? Property on what type?). This makes both issues unfixable.

The only thing to do is follow @jpsim’s advice,

> [B]e aware of the false‐positives, and don’t use opt‐in rules unless you’re aware of their tradeoffs and find them acceptable.

Please know that we marked this as a duplicate to help organize the list of issues, not to label you as too lazy to search. ;) We *are* thankful for your report.

**Comment by yas375** [comment]
Thanks Jeremy 👍

**Comment by yo1995** [comment]
FYI: I run into the same linter error when checking against `NSDictionary.count`. Switching to `NSDictionary.allKeys.isEmpty` workaround the problem.
